notebook/gateway/managers.py

```diff
@@ -540,9 +551,32 @@ def get_kernel_spec(self, kernel_name, **kwargs):
                 raise
         else:
             kernel_spec = json_decode(response.body)
-            # Convert to instance of Kernelspec
-            kspec_instance = self.kernel_spec_class(resource_dir=u'', **kernel_spec['spec'])
-        raise gen.Return(kspec_instance)
+
+        raise gen.Return(kernel_spec)
+
+    @gen.coroutine
+    def get_kernel_spec_resource(self, kernel_name, path):
+        """Get kernel spec for kernel_name.
+
+        Parameters
+        ----------
+        kernel_name : str
+            The name of the kernel.
+        path : str
+            The name of the desired resource
+        """
+        kernel_spec_resource_url = url_path_join(self.base_resource_endpoint, str(kernel_name), str(path))
+        self.log.debug("Request kernel spec resource '{}' at: {}".format(path, kernel_spec_resource_url))
+        try:
+            response = yield gateway_request(kernel_spec_resource_url, method='GET')
+        except HTTPError as error:
+            if error.code == 404:
+                kernel_spec_resource = None
+            else:
+                raise
+        else:
+            kernel_spec_resource = response.body
+        raise gen.Return(kernel_spec_resource)
```

Comment:
FWIW, since we require Python 3, we can use `return kernel_spec_resource`

Reply:
Ok - thanks.  Since this is one of many gen.Returns, I think it makes sense to change these whenever we "make the sweep" to move to async/await - if that's okay with you.